backend/models/ticket.js

```diff
@@ -231,34 +231,49 @@ class Ticket {
 		const systemComments = [];
 		const fields = Object.keys(data);
 
-		fields.forEach(field => {
-			if (Object.prototype.toString.call(data[field]) !== this.fieldTypes[field]) {
-				throw responseCodes.INVALID_ARGUMENTS;
-			}
+		let newViewpoint;
 
-			// if a field have the same value shouldnt update the property
-			if (_.isEqual(oldTicket[field], data[field])) {
-				delete data[field];
+		fields.forEach(async field => {
```

Comment:
Why is this an async? It doesn't look like there's anything to await for?

If this is async then we need to collect the promises and wait on them, otherwise we may run into race conditions